### DISCUSSION TITLE
resource script (/opt/resource/out /tmp/build/put) output a null version

### DISCUSSION THREAD

**Discussion by kalrashubham49** [concourse/concourse#6239] [Help & Support]
## Summary
Custom Resource Not Working

## Steps to reproduce
when trying to run a pipeline with below content out executable is not working

```yaml
resource_types:
    - name: http-api
      type: docker-image
      source:
        repository: kalrashubham49/custom-http-resource
        tag: v2.2

resources:
      - name: clair
        type: http-api
        source:
            uri: http://45.33.90.33:8081/health
            method: GET
jobs:
      - name: test-http
        plan:
          - put: clair
```
            

## Triaging info

* Concourse version:
* Browser (if applicable):
* Did this used to work?

**Comment by jamieklassen** [comment]
tl;dr - please provide more details, as there's very little information to help with.

Converting to a discussion as we have no reproducible. It sounds like you're looking for advice on how to write a custom resource type, but to get meaningful help you're probably going to need to share:

* a reproducible concourse setup, ideally in the form of a `docker-compose.yml` file so that it's clear what version of concourse you're running and how it's configured.
* the code that runs in your `kalrashubham49/custom-http-resource` image. I looked around on your github profile but I didn't see anything obvious.

I'll mention https://concourse-ci.org/implementing-resource-types.html, but it seems like you've probably read that.

The error you're getting is from

https://github.com/concourse/concourse/blob/248606d42c177df7da97c65addd3e5955ce03bc4/atc/resource/resource_put.go#L36

which suggests that your `/opt/resource/out` script is printing valid JSON on stdout, but that JSON is supposed to have a structure like

```json
{
  "version": { "ref": "61cebf" },
  "metadata": [
    { "name": "commit", "value": "61cebf" },
    { "name": "author", "value": "Mick Foley" }
  ]
}
```

In your case, when the output is parsed, the result has a null value for `version`. This suggests either that your script is outputting JSON like `{"version":null}` or more likely that your script is outputting JSON that doesn't have a `version` field at all.

EDIT: I just tried

```
$ echo '{"source":{"uri":"http://45.33.90.33:8081/health","method":"GET"}}' | docker run --rm kalrashubham49/custom-http-resource:v2.2 /opt/resource/out /tmp
{"ref":""}
```

which suggests pretty strongly that your script needs to be printing JSON with a `version` field in it.
